java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -43,14 +47,43 @@
   private Path workspace = null;
 
   public static void setUpClass() throws IOException {
+    String configFile = System.getProperty("bazel.configuration");
+    properties = new Properties();
+    try (InputStream is = new FileInputStream(configFile)) {
+      properties.load(is);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
     String environmentTempDirectory = System.getenv("TEST_TMPDIR");
     if (environmentTempDirectory == null) {
       tmp = Files.createTempDirectory("e4b-tests");
       tmp.toFile().deleteOnExit();
     } else {
       tmp = Paths.get(environmentTempDirectory);
     }
+
     bazelVersions = new HashMap<>();
+
+    // Create the distdir with the external dependency archives
+    String externalDeps = properties.getProperty("bazel.external.deps");
+    if (externalDeps != null && !externalDeps.isEmpty()) {
+      Path distdir = tmp.resolve("distdir");
+      Files.createDirectories(distdir);
+      try (Stream<Path> extDeps = Files.list(distdir)) {
+        extDeps.forEach(path -> {
+          try {
+            Files.delete(path);
```

Comment:
won't this fail if the `path` represents a directory?

Reply:
It should not really be possible, though, because that's not how distdir is set to work (it is a flat dir by design), and if you have a subdirectory, then something is very, very broken.